tests/test_game.py

```diff
@@ -0,0 +1,249 @@
+
+    assert isinstance(game.game_start, bool)
+    assert isinstance(game.game_complete, bool)
+    assert isinstance(game.finalized, bool)
+
+    assert isinstance(game.inning, int)
+    assert isinstance(game.top_of_inning, bool)
+    assert isinstance(game.home_score, (int, float))
+    assert isinstance(game.away_score, (int, float))
+    assert isinstance(game.winning_score, (int, float))
+    assert isinstance(game.losing_score, (int, float))
+    assert isinstance(game.half_inning_score, (int, float))
+
+    assert isinstance(game.half_inning_outs, int)
+    assert isinstance(game.baserunner_count, int)
+    assert isinstance(game.at_bat_balls, int)
+    assert isinstance(game.at_bat_strikes, int)
+    assert isinstance(game.home_team_batter_count, int)
+    assert isinstance(game.away_team_batter_count, int)
+
+
+def test_mods(game):
+    """
+    Test that game modifications do not error
+    """
+    assert isinstance(game.base_runner_mods, list)
+    for mod in game.base_runner_mods:
+        assert isinstance(mod, Modification)
+    assert isinstance(game.home_pitcher_mod, (Modification, type(None)))
+    assert isinstance(game.away_pitcher_mod, (Modification, type(None)))
+    assert isinstance(game.home_batter_mod, (Modification, type(None)))
+    assert isinstance(game.away_batter_mod, (Modification, type(None)))
+
+
+def test_weather(game):
+    """
+    Test that weather does not error
+    """
+    assert isinstance(game.weather, Weather)
+    if game.weather.value != None:
+        assert game.weather.text != "Invalid Weather"
+
+
+def test_teams(game):
+    """
+    Test that teams & team fields do not error
+    """
+    assert isinstance(game.away_team, Team)
+    assert isinstance(game.home_team, Team)
+    assert isinstance(game.winning_team, Team)
+    assert isinstance(game.losing_team, Team)
+    assert isinstance(game.losing_team, Team)
+    assert isinstance(game.at_bat_team, Team)
+    assert isinstance(game.pitching_team, Team)
+
+    assert isinstance(game.away_team_name, str)
+    assert isinstance(game.away_team_nickname, str)
+    assert isinstance(game.home_team_name, str)
+    assert isinstance(game.home_team_nickname, str)
+    assert isinstance(game.winning_team_name, str)
+    assert isinstance(game.winning_team_nickname, str)
+    assert isinstance(game.losing_team_name, str)
+    assert isinstance(game.losing_team_nickname, str)
+    assert isinstance(game.at_bat_team_name, str)
+    assert isinstance(game.at_bat_team_nickname, str)
+    assert isinstance(game.pitching_team_name, str)
+    assert isinstance(game.pitching_team_nickname, str)
+
+    assert isinstance(game.away_team_color, str)
+    assert isinstance(game.away_team_emoji, str)
+    assert isinstance(game.home_team_color, str)
+    assert isinstance(game.home_team_emoji, str)
+
+
+def test_players(game):
+    """
+    Test that players & player fields do not error
+    """
+    assert isinstance(game.away_pitcher, (Player, type(None)))
```

Comment:
this is a pretty good example of where explicit fixtures and parametrization would make this more robust. `None` is not a correct pitcher value for basically any valid game in the past, and not `None` isn't necessarily correct for any game in the future.

Reply:
Your reasoning is correct but hilariously enough `game.awayPitcher` is actually None in some cases https://www.blaseball.com/database/gameById/50a6fc92-acdd-4999-a1c1-1e5e1eedc685